### PR TITLE
Add ES5+CommonJS support

### PR DESCRIPTION
**Pull request by beck**
Compile all the source at prepublish.

Motivation here is to test a consumer using node.

ATM, a consumer's test will fail on the ESM syntax (which we workaround by
browswerifying, making unit tests very slow).

The cost to adding CommonJS support is that the module can only have
one export and is why the member exports have been removed.

### CODE REVIEW

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

File: scripts/compile-common-js.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+proj=$(dirname "$0")/..
+cd "$proj"
+
+rm -rf dist
+
+echo "Compiling to ES5+CommonJS ..."
+for f in $(find src -name "*.js")
+do
+    if [[ $f == *"-spec.js"* ]]; then
```
Comment (stefanor) [inline scripts/compile-common-js.sh]: Considered adding `! -name '*-spec.js'` to the `find`?

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

File: scripts/compile-common-js.sh [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+set -eu
+
+proj=$(dirname "$0")/..
+cd "$proj"
+
+rm -rf cjs
+
+echo "Compiling to ES5+CommonJS ..."
+for f in $(find src -name "*.js" ! -name '*-spec.js')
```
Comment (stefanor) [inline scripts/compile-common-js.sh]: Consistent quotes?

Comment (beck) [reply scripts/compile-common-js.sh]: Shame on me for copy/pasting your critique

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

File: scripts/compile-common-js.sh (hunk 2 above)
Comment (stefanor) [reply scripts/compile-common-js.sh]: Yeah, I realised that I'm to blame for that :P

**Review by stefanor — APPROVED** [review]

**Pull request merged**